conn.go

```diff
@@ -354,23 +356,38 @@ func processLoop(c *Conn, writer *frame.Writer) {
 				}
 			}
 
+			// default is to always send a frame.
+			var sendFrame = true
+
 			switch req.Frame.Command {
 			case frame.SUBSCRIBE:
 				id, _ := req.Frame.Header.Contains(frame.Id)
 				channels[id] = req.C
+
+				// if using a temp queue, map that destination as a known channel
+				// however, don't send the frame, it's most likely an invalid destination
+				// on the broker.
+				if replyTo, ok := req.Frame.Header.Contains(ReplyToHeader); ok {
+					channels[replyTo] = req.C
+					sendFrame = false
```

Comment:
we can `break`/`return` here instead of using a flag to prevent the frame being sent

Reply:
Returning here actually stops anything being sent at all, when I replace it with a return the entire `processLoop` stops operating. a `break` will still always send the frame as it sits outside the switch.

Reply:
You're right, I forgot completely about the for loop processing the channel.